share/mk/src.opts.mk

```diff
@@ -368,6 +368,14 @@ BROKEN_OPTIONS+=GOOGLETEST SSP
 BROKEN_OPTIONS+=NS_CACHING
 .endif
 
+.if ${__T} == "morello"
+# Transitively includes pcpu.h pcpu_aux.h and fails on the:
```

Comment:
Where are they being pulled in? They should only be used in the kernel, with the latter checking for `_KERNEL`

Reply:
Yeah I was confused too, but it was this:
```
In file included from /Users/Jess/cheri/morello-cheribsd/lib/libprocstat/zfs_defs.c:53:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h:41:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/debug.h:33:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/systm.h:35:
In file included from /Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/sys/systm.h:149:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/pcpu.h:33:
In file included from /Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/sys/pcpu.h:50:
/Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/machine/pcpu.h:63:41: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
        __asm __volatile("mov   %0, x18" : "=&r"(pcpu));
                                                 ^
/Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/machine/pcpu.h:63:24: note: use constraint modifier "w"
        __asm __volatile("mov   %0, x18" : "=&r"(pcpu));
                                ^
/Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/machine/pcpu.h:72:43: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
        __asm __volatile("ldr   %0, [x18]" : "=&r"(td));
                                                   ^
/Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/machine/pcpu.h:72:24: note: use constraint modifier "w"
        __asm __volatile("ldr   %0, [x18]" : "=&r"(td));
                                ^
In file included from /Users/Jess/cheri/morello-cheribsd/lib/libprocstat/zfs_defs.c:53:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h:41:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/debug.h:33:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/systm.h:35:
In file included from /Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/sys/systm.h:149:
In file included from /Users/Jess/cheri/morello-cheribsd/sys/cddl/compat/opensolaris/sys/pcpu.h:33:
In file included from /Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/sys/pcpu.h:224:
/Users/Jess/cheri/build/cheribsd-morello-purecap-build/Users/Jess/cheri/morello-cheribsd/arm64.morello/tmp/usr/include/machine/pcpu_aux.h:48:1: error: static_assert failed due to requirement '(1 << 12) % sizeof(struct pcpu) == 0' "fix pcpu size"
_Static_assert(PAGE_SIZE % sizeof(struct pcpu) == 0, "fix pcpu size");
^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
--- zfs_defs.o ---
```

zfs_defs.c is a terrible thing and has:
```c
/* Pretend we are kernel to get the same binary layout. */
#define _KERNEL

/* A hack to deal with kpilite.h. */
#define KLD_MODULE

/*
 * Prevent some headers from getting included and fake some types
 * in order to allow this file to compile without bringing in
 * too many kernel build dependencies.
 */
#define _OPENSOLARIS_SYS_PATHNAME_H_
#define _OPENSOLARIS_SYS_POLICY_H_
#define _OPENSOLARIS_SYS_VNODE_H_
#define _VNODE_PAGER_

typedef struct vnode vnode_t;
typedef struct vattr vattr_t;
typedef struct xvattr xvattr_t;
typedef struct vsecattr vsecattr_t;
typedef enum vtype vtype_t;

#include <sys/zfs_context.h>
#include <sys/zfs_znode.h>

size_t sizeof_znode_t = sizeof(znode_t);
size_t offsetof_z_id = offsetof(znode_t, z_id);
size_t offsetof_z_size = offsetof(znode_t, z_size);
size_t offsetof_z_mode = offsetof(znode_t, z_mode);

/* Keep pcpu.h satisfied. */
uintptr_t *__start_set_pcpu;
uintptr_t *__stop_set_pcpu;
```
Hopefully OpenZFS has a better interface to include from userspace. Also the "Pretend we are kernel to get the same binary layout." is of course broken if you mix kernel and userspace ABIs............